logging/servicelogger.go

```diff
@@ -3,176 +3,276 @@ package logging
 import (
 	"fmt"
 	"io"
-	"log"
+	"net/http"
+	"os"
+	"strings"
+
+	"github.com/tozny/utils-go"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
 )
 
-// ServiceLogger represents a logger with logging level prefixes for a specific service.
 type ServiceLogger struct {
-	logLevel    int
+	logConfig   *zap.Config
 	serviceName string
-	*log.Logger
-}
-
-var levelMap = map[string]int{
-	"OFF":      0,
-	"SERVICE":  1,
-	"CRITICAL": 2,
-	"ERROR":    3,
-	"INFO":     4,
-	"DEBUG":    5,
+	*zap.SugaredLogger
 }
 
-// NewServiceLogger returns a logger with designated logging levels for a particular service.
+// NewZapSugaredServiceLogger returns a logger with designated logging levels for a particular service.
 func NewServiceLogger(out io.Writer, serviceName string, level string) ServiceLogger {
+	var sugaredZapLogger *zap.SugaredLogger
+	// Get a default configuration
+	config := zap.NewProductionConfig()
+	config.OutputPaths = []string{"stdout"}
+	// Set logging level
+	config.Level.SetLevel(zapLevelMap[level])
+	config.EncoderConfig.NameKey = "service"
+	config.EncoderConfig.MessageKey = "message"
+
+	initialFields := make(map[string]interface{}, 1)
+
+	initialFields["facility"] = "user-level"
+	config.InitialFields = initialFields
+	config.EncoderConfig.EncodeLevel = zapcore.CapitalLevelEncoder
+	config.EncoderConfig.EncodeTime = zapcore.ISO8601TimeEncoder
+
+	zapLogger, err := config.Build(zap.WithCaller(false))
+	if err != nil {
+		panic(fmt.Errorf("Logger could not be built. This is not an expected outcome. ERR: %+v", err))
+	}
+	var loggingFormat = utils.MustGetenv("LOGGING_FORMAT")
+	var enc zapcore.Encoder
+	if strings.EqualFold("Syslog", loggingFormat) {
+		enc = NewSyslogEncoder(SyslogEncoderConfig{
+			EncoderConfig: zapcore.EncoderConfig{
+				NameKey:        "logger",
+				CallerKey:      "caller",
+				MessageKey:     "msg",
+				StacktraceKey:  "stacktrace",
+				EncodeLevel:    zapcore.LowercaseLevelEncoder,
+				EncodeTime:     zapcore.EpochTimeEncoder,
+				EncodeDuration: zapcore.SecondsDurationEncoder,
+				//EncodeCaller:   zapcore.ShortCallerEncoder,
```

Comment:
what does this do ?

Reply:
These are some in built functionalities of zapcore to customise the logging format. LowerCaseLevelEncoder - to display logging level(INFO,DEBUG,..) in lowercase. EpochTimeEncoder - time serialization.